Configuration/PyReleaseValidation/python/relval_steps.py

```diff
@@ -1745,8 +1761,13 @@ def lhegensim2018ml(fragment,howMuch):
 
 steps['DIGIPRMXUP15_PU50']=merge([digiPremixUp2015Defaults50ns])
 steps['DIGIPRMXUP17_PU25']=merge([digiPremixUp2017Defaults25ns])
+steps['DIGIPRMXUP17_PU25_RD']=merge([digiPremixUp2017Defaults25ns, { '--customise_commands':"\"process.EcalLaserCorrectionServiceMC = cms.ESProducer('EcalLaserCorrectionServiceMC') \\n process.GlobalTag.toGet = cms.VPSet( cms.PSet(     record = cms.string('EcalLaserAPDPNRatiosMCRcd'),  tag = cms.string('EcalLaserAPDPNRatios_UL_2017_mc'),  connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS') ) )  \\n process.ecal_sim_parameter_map.timeDependent=cms.bool(True)\"" } ])
 steps['DIGIPRMXLOCALUP17_PU25']=merge([digiPremixLocalPileupUp2017Defaults25ns])
 steps['DIGIPRMXUP18_PU25']=merge([digiPremixUp2018Defaults25ns])
+#
+steps['DIGIPRMXUP18_PU25_RD_test']=merge([digiPremixUp2018Defaults25ns, { '--customise_commands':"\"process.EcalLaserCorrectionServiceMC = cms.ESProducer('EcalLaserCorrectionServiceMC') \\n process.GlobalTag.toGet = cms.VPSet( cms.PSet(     record = cms.string('EcalLaserAPDPNRatiosMCRcd'),  tag = cms.string('EcalLaserAPDPNRatios_Run_Dep_MC_first_IOV'),  connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS') ), cms.PSet(     record = cms.string('EcalLaserAPDPNRatiosRcd'),  tag = cms.string('EcalLaserAPDPNRatios_Run_Dep_MC'),  connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS') ) )  \\n process.ecal_sim_parameter_map.timeDependent=cms.bool(True) \\n process.source.setRunNumberForEachLumi = cms.untracked.vuint32(315257,316083)\"" } ])
+steps['DIGIPRMXUP18_PU25_RD']=merge([digiPremixUp2018Defaults25ns, { '--customise_commands':"\"process.EcalLaserCorrectionServiceMC = cms.ESProducer('EcalLaserCorrectionServiceMC') \\n process.GlobalTag.toGet = cms.VPSet( cms.PSet(     record = cms.string('EcalLaserAPDPNRatiosMCRcd'),  tag = cms.string('EcalLaserAPDPNRatios_Run_Dep_MC_first_IOV'),  connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS') ), cms.PSet(     record = cms.string('EcalLaserAPDPNRatiosRcd'),  tag = cms.string('EcalLaserAPDPNRatios_Run_Dep_MC'),  connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS') ) )  \\n process.ecal_sim_parameter_map.timeDependent=cms.bool(True) \\n process.source.setRunNumberForEachLumi = cms.untracked.vuint32((315257,)*20 + (316082,)*20 + (316720,)*20 + (317527,)*20 + (320917,)*20 + (321414,)*20 + (321973,)*20 + (322492,)*20 + (324245,)*20)\"" } ])
```

Comment:
I started to wonder if
```python
process.ecal_sim_parameter_map.timeDependent=cms.bool(True)
```
customization actually does what it is expected to do?

As far as I can tell, `ecal_sim_parameter_map` is a PSet defined here 
https://github.com/cms-sw/cmssw/blob/6027981a5c10a3f8986ce4b989628923125b5d34/SimCalorimetry/EcalSimProducers/python/ecalSimParameterMap_cff.py#L3
(and modified in this PR), and the PSet gets propagated to `MixingModule` and `PreMixingModule` via
https://github.com/cms-sw/cmssw/blob/6027981a5c10a3f8986ce4b989628923125b5d34/SimGeneral/MixingModule/python/ecalDigitizer_cfi.py#L11-L16
This fragment, however, *copies* the `ecal_sim_parameter_map` PSet, and therefore any modifications done to `process.ecal_sim_parameter_map` after the `process.load(...)` of `(Pre)MixingModule` do not propagate to `(Pre)MixingModule` (caveat: I didn't check the configurations generated from this PR, but the copy behavior is easy to test e.g. in python shell).

Which one of `MixingModule` (hard-scatter digi accumulation) and `PreMixingModule` (pileup overlay and `finalizeEvent()`) should actually use the value `True`?

Reply:
> I started to wonder if
> 
> ```python
> process.ecal_sim_parameter_map.timeDependent=cms.bool(True)
> ```
> 
> customization actually does what it is expected to do?
> 
> As far as I can tell, `ecal_sim_parameter_map` is a PSet defined here
> https://github.com/cms-sw/cmssw/blob/6027981a5c10a3f8986ce4b989628923125b5d34/SimCalorimetry/EcalSimProducers/python/ecalSimParameterMap_cff.py#L3
> 
> 
> (and modified in this PR), and the PSet gets propagated to `MixingModule` and `PreMixingModule` via
> https://github.com/cms-sw/cmssw/blob/6027981a5c10a3f8986ce4b989628923125b5d34/SimGeneral/MixingModule/python/ecalDigitizer_cfi.py#L11-L16
> 
> 
> This fragment, however, _copies_ the `ecal_sim_parameter_map` PSet, and therefore any modifications done to `process.ecal_sim_parameter_map` after the `process.load(...)` of `(Pre)MixingModule` do not propagate to `(Pre)MixingModule` (caveat: I didn't check the configurations generated from this PR, but the copy behavior is easy to test e.g. in python shell).
> Which one of `MixingModule` (hard-scatter digi accumulation) and `PreMixingModule` (pileup overlay and `finalizeEvent()`) should actually use the value `True`?


The "timeDependent" should be set to true when mixing the premixed pileup library with the hard scattering.

Is this not happening with the customization as defined in the workflow definition for runthematrix?

Reply:
> The "timeDependent" should be set to true when mixing the premixed pileup library with the hard scattering.

Thanks (so `PreMixingModule`, and ok, it is actually evident from the code of this PR as well).

> Is this not happening with the customization as defined in the workflow definition for runthematrix?

Judging by eye, no (but could useful to double-check). I believe the proper customization line should be
```python
process.mixData.workers.ecal.timeDependent=True
```

Reply:
You are correct. I tested it explicitly with edmconfigdump and the "timedependent" set to true is not propagated.

I will push this fix in the commit.